### PR TITLE
[PAY-1145] DMs: Add InboxUnavailableModal

### PR DESCRIPTION
**Pull request by rickyrombo**
### Description

Adds a new modal for when the user tries to click the message icon on a profile that has messages disabled for whatever reason.

When i blocked the user I'm trying to send a message to:
image

When the user has made DMs tip gated and I haven't tipped:
image

When the user turned off their inbox or has blocked me:
image

### Dragons

*Is there anything the reviewer should be on the lookout for? Are there any dangerous changes?*

### How Has This Been Tested?

*Please describe the tests that you ran to verify your changes. Provide repro instructions & any configuration.*

### How will this change be monitored?

*For features that are critical or could fail silently please describe the monitoring/alerting being added.*

### Feature Flags ###

*Are all new features properly feature flagged? Describe added feature flags.*

### CODE REVIEW

File: packages/web/src/components/inbox-unavailable-modal/InboxUnavailableModal.module.css
Comment (rickyrombo) [inline packages/web/src/components/inbox-unavailable-modal/InboxUnavailableModal.module.css:22]: FYI @dylanjeffers a good case study re: Harmony, I think this file should be made wholly unnecessary, but especially the `.content`, `.footer` and `.icon` classes

File: packages/web/src/components/inbox-unavailable-modal/InboxUnavailableModal.tsx
Comment (rickyrombo) [inline packages/web/src/components/inbox-unavailable-modal/InboxUnavailableModal.tsx:139]: Waiting for Design / Product decision here, will remove before merging otherwise

File: packages/web/src/components/inbox-unavailable-modal/InboxUnavailableModal.module.css
Comment (dylanjeffers) [reply packages/web/src/components/inbox-unavailable-modal/InboxUnavailableModal.module.css:22]: Thanks for heads up! What would you prefer to see instead? Having a box component that handles this, or using flex gap or something?

File: packages/web/src/pages/profile-page/ProfilePageProvider.tsx
Comment (rickyrombo) [inline packages/web/src/pages/profile-page/ProfilePageProvider.tsx:273]: RE: modals @sliptype / @dylanjeffers / @schottra - Do we hate this?

(Modal mounted in desktop version below)

File: packages/web/src/components/inbox-unavailable-modal/InboxUnavailableModal.module.css
Comment (rickyrombo) [reply packages/web/src/components/inbox-unavailable-modal/InboxUnavailableModal.module.css:22]: None of the above - all of these styles could easily be made as defaults to the relevant modal components

This is such a basic modal that it should just work out of the box

Comment (rickyrombo) [reply packages/web/src/components/inbox-unavailable-modal/InboxUnavailableModal.module.css:22]: Eg. Button should have a variant that's full width, ModalFooter should have no top padding but the same horizontal and bottom padding as ModalContent, and ModalContent should have the proper font style defaults. Icons in the ModalHeader should fill with the appropriate colors (whether that's at the icon level or the modal level idk) and perhaps the modal width should be a prop?

**Comment by audius-infra** [comment]
Preview this change https://demo.audius.co/mjp-dms-permit-fix

File: packages/web/src/components/inbox-unavailable-modal/InboxUnavailableModal.module.css
Comment (schottra) [reply packages/web/src/components/inbox-unavailable-modal/InboxUnavailableModal.module.css:22]: You could also make the case (a la MUI) that `ModalHeader` and `ModalFooter` could be reusable components that render the common cases for us and just accept callbacks for things like clicking the primary action etc.

Comment (dylanjeffers) [reply packages/web/src/components/inbox-unavailable-modal/InboxUnavailableModal.module.css:22]: Oh yeah 💯

**Comment by audius-infra** [comment]
Preview this change https://demo.audius.co/mjp-dms-permit-fix

File: packages/common/src/store/pages/chat/selectors.ts
Comment (dharit-tan) [inline packages/common/src/store/pages/chat/selectors.ts:213]: noting here that we default to user is not permitted until we load in the permissions

Comment (rickyrombo) [reply packages/common/src/store/pages/chat/selectors.ts:213]: yeah.. added a CTA that's "WAIT" so we can handle that case, but yeah it'll quickly flip from not allowed to allowed

File: packages/web/src/components/inbox-unavailable-modal/InboxUnavailableModal.tsx
Comment (dharit-tan) [inline packages/web/src/components/inbox-unavailable-modal/InboxUnavailableModal.tsx:97]: Is it a bit weird to have this be clickable to navigate to the user's profile, when we're already on the profile page?

**Review by dharit-tan — APPROVED** [review]

File: packages/web/src/components/inbox-unavailable-modal/InboxUnavailableModal.tsx
Comment (rickyrombo) [reply packages/web/src/components/inbox-unavailable-modal/InboxUnavailableModal.tsx:97]: Yeah.. but I think we should keep for consistency

**Pull request merged**
